com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ModuleDependencyChecker.java

```diff
@@ -0,0 +1,91 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.Module;
+
+public class ModuleDependencyChecker {
+
+    private Set<String> modulesIncludedInDeploy;
+    private Set<String> modulesExcludedFromDeploy;
+    private Set<String> modulesAlreadyDeployed;
+    CloudControllerClient client;
+
+    public ModuleDependencyChecker(CloudControllerClient client, DeploymentDescriptor descriptor, List<Module> includedModules,
+        List<Module> completedModules) {
+        this.client = client;
+
+        modulesIncludedInDeploy = includedModules.stream()
+            .map(Module::getName)
+            .collect(Collectors.toSet());
+        modulesAlreadyDeployed = completedModules.stream()
+            .map(Module::getName)
+            .collect(Collectors.toSet());
+        modulesExcludedFromDeploy = computeLeftOutModules(descriptor);
+    }
+
+    private Set<String> computeLeftOutModules(DeploymentDescriptor deploymentDescriptor) {
+
+        Set<String> leftOutModules = deploymentDescriptor.getModules()
+            .stream()
+            .map(Module::getName)
+            .filter(module -> {
+                return !modulesIncludedInDeploy.contains(module);
+            })
+            .collect(Collectors.toSet());
+
+        return leftOutModules;
+    }
+
+    public boolean allDependenciesAreSatisfied(Module module) {
+        if (module.getMajorSchemaVersion() < 3) {
+            return true;
+        }
+
+        return module.getDeployedAfter()
+            .isEmpty() || modulesAlreadyDeployed.containsAll(module.getDeployedAfter())
+            || allDependenciesAreAlreadyPresent(module.getDeployedAfter());
+    }
+
+    private boolean allDependenciesAreAlreadyPresent(List<String> deployedAfter) {
+        List<CloudApplication> appsLeftOutOfDeploymentNotFoundInSpace = deployedAfter.stream()
```

Comment:
Huge name. Can we split it in two?
You can have two streams, one depending on the other. That way it will be more readable.